hazelcast-jet-core/src/main/java/com/hazelcast/jet/pipeline/Sources.java

```diff
@@ -1012,6 +1057,50 @@ public static FileSourceBuilder filesBuilder(@Nonnull String directory) {
         return filesBuilder(watchedDirectory).buildWatcher();
     }
 
+    /**
+     * A source to stream lines added to files with a `.json` extension in a
+     * directory. This is a streaming source, it will watch directory and emit
+     * objects of given {@code type} by converting each line as they are
+     * appended to files in that directory.
+     * <p>
+     * This method is a shortcut for: <pre>{@code
+     *   filesBuilder(directory)
+     *      .charset(UTF_8)
+     *      .glob(GLOB_WILDCARD)
+     *      .sharedFileSystem(false)
+     *      .buildWatcher((JsonUtil.asJson(type))
+     * }</pre>
+     *
+     * <h3>Appending lines using an text editor</h3>
+     * If you're testing this source, you might think of using a text editor to
+     * append the lines. However, it might not work as expected because some
+     * editors write to a temp file and then rename it or append extra newline
+     * character at the end which gets overwritten if more text is added in the
+     * editor. The best way to append is to use {@code echo text >> yourFile}.
+     *
+     * See {@link #filesBuilder(String)}, {@link #fileWatcher(String)},
+     * and {@link JsonUtil#asJson(Class)}.
+     *
+     * @since 4.2
+     */
+    @Nonnull
+    public static <T> StreamSource<T> jsonWatcher(@Nonnull String watchedDirectory, @Nonnull Class<T> type) {
```

Comment:
I don't see any test coverage for this, does it actually work as intended? or we can get half-written JSON?